### PR TITLE
Optional miner ID for k2pow input

### PR DESCRIPTION
**Pull request by poszu**
Part of https://github.com/spacemeshos/go-spacemesh/issues/4656

### CODE REVIEW

File: Makefile.Inc [hunk 1]
```diff
@@ -49,7 +49,7 @@ else
 	endif
 endif
 
-POSTRS_SETUP_REV = 0.3.1
+POSTRS_SETUP_REV = 0.4.0-rc2
```
Comment (poszu) [inline Makefile.Inc]: TODO: Update when post-rs is merged and released

File: internal/postrs/proof.go [hunk 2]
```diff
@@ -34,7 +35,26 @@ func TranslateScryptParams(n, r, p uint) ScryptParams {
...
 }
 
-func GenerateProof(dataDir string, challenge []byte, logger *zap.Logger, nonces, threads uint, K1, K2 uint32, powDifficulty [32]byte, powFlags PowFlags) (*shared.Proof, error) {
+type postOptions struct {
+	powCreatorId []byte
+}
+
+type PostOptionFunc func(*postOptions) error
+
+func WithPowCreator(id []byte) PostOptionFunc {
```
Comment (fasmat) [inline internal/postrs/proof.go:44]: Depending on the effort, would it make sense to change all the other properties also to functional options?

File: internal/postrs/proof.go [hunk 3]
```diff
@@ -138,7 +166,14 @@ func (v *Verifier) VerifyProof(
 	k1, k2, k3 uint32,
 	powDifficulty [32]byte,
 	scryptParams ScryptParams,
+	options ...PostOptionFunc,
```
Comment (fasmat) [inline internal/postrs/proof.go:169]: Same as for proof generation, would it make sense to change the other parameters too?

File: internal/postrs/proof.go [hunk 4]
```diff
@@ -183,6 +218,16 @@ func (v *Verifier) VerifyProof(
 		},
 	}
 
+	if opts.powCreatorId != nil {
+		logger.With().Info("Verifying with PoW creator ID", zap.String("id", hex.EncodeToString(opts.powCreatorId)))
+		minerIdSliceHdr := (*reflect.SliceHeader)(unsafe.Pointer(&opts.powCreatorId))
```
Comment (fasmat) [inline internal/postrs/proof.go:223]: Why here not also just `C.Bytes(opts.PowCraetorId)` and free it on the go side afterwards?

`len` is always guaranteed to be 32 already checked before passed into this function and `cap` should be irrelevant for rust.

File: internal/postrs/proof.go (hunk 2 above)
Comment (poszu) [reply internal/postrs/proof.go:44]: Yes, it makes sense. But it's an effort we don't have time for right now.

File: internal/postrs/proof.go (hunk 4 above)
Comment (poszu) [reply internal/postrs/proof.go:223]: I followed the logic above already present for indices. It's nice as, in contrast to `C.Bytes()` it doesn't copy.

**Review by fasmat — APPROVED** [review]

File: internal/postrs/proof.go (hunk 2 above)
Comment (fasmat) [reply internal/postrs/proof.go:44]: Created issue to address later: https://github.com/spacemeshos/post/issues/174

**Pull request merged**